core/src/test/java/org/apache/iceberg/expressions/TestInclusiveMetricsEvaluatorWithExtract.java

```diff
@@ -683,4 +784,1638 @@ public void testIntegerNotIn() {
+        .isTrue();
+    assertThat(
+            shouldRead(
+                notEqual(
+                    extract(
+                        "variant",
+                        "$.event_timestamp_long",
+                        Types.TimestampType.withZone().toString()),
+                    TS_COMMON_MAX_VALUE_MINUS_10.toString())))
+        .as("Should read: timestamp between lower and  upper bound")
+        .isTrue();
+    assertThat(
+            shouldRead(
+                notEqual(
+                    extract(
+                        "variant",
+                        "$.event_timestamp_long",
+                        Types.TimestampType.withZone().toString()),
+                    TS_COMMON_MAX_VALUE.toString())))
+        .as("Should read: timestamp equal to upper bound")
+        .isTrue();
+
+    // Timestamp as timestamp data type
+    assertThat(
+            shouldRead(
+                notEqual(
+                    extract(
+                        "variant",
+                        "$.event_timestamp_ts_type",
+                        Types.TimestampType.withZone().toString()),
+                    TS_COMMON_MIN_VALUE_MINUS_10.toString())))
+        .as("Should read: timestamp below lower bound")
+        .isTrue();
+    assertThat(
+            shouldRead(
+                notEqual(
+                    extract(
+                        "variant",
+                        "$.event_timestamp_ts_type",
+                        Types.TimestampType.withZone().toString()),
+                    TS_COMMON_MIN_VALUE.toString())))
+        .as("Should read: timestamp equal to lower bound")
+        .isTrue();
+    assertThat(
+            shouldRead(
+                notEqual(
+                    extract(
+                        "variant",
+                        "$.event_timestamp_ts_type",
+                        Types.TimestampType.withZone().toString()),
+                    TS_COMMON_MAX_VALUE_MINUS_10.toString())))
+        .as("Should read: timestamp between lower and upper bounds")
+        .isTrue();
+    assertThat(
+            shouldRead(
+                notEqual(
+                    extract(
+                        "variant",
+                        "$.event_timestamp_ts_type",
+                        Types.TimestampType.withZone().toString()),
+                    TS_COMMON_MAX_VALUE_PLUS_10.toString())))
+        .as("Should read: timestamp above upper bound")
+        .isTrue();
+    assertThat(
+            shouldRead(
+                notEqual(
+                    extract(
+                        "variant",
+                        "$.event_timestamp_ts_type",
+                        Types.TimestampType.withZone().toString()),
+                    TS_COMMON_MAX_VALUE.toString())))
+        .as("Should read: timestamp equal to upper bound")
+        .isTrue();
+  }
+
+  @Test
+  public void testTimestampNanoLt() {
+    assertThat(
+            shouldRead(
```

Comment:
Instead of duplicate the same tests, can you take a look if we can use parameterized tests with 

"$.event_timestamp_nano_long", TS_NANO_MIN_VALUE, L_TS_NANO_MIN_VALUE_MINUS_10, etc as the inputs? 

Otherwise, they are the same as micro_timestamp, right?